### PR TITLE
Ph add dbgap status wdl

### PR DESCRIPTION
**Pull request by phendriksen100**
- Created wdl to allow us to check the status of all samples in DbGap
- Updated telemetry report wrapper to use xmlToDict instead of the ET module

### CODE REVIEW

File: src/services/dbgap_telemetry_report.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        self.endpoint = f"https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetSampleStatus.cgi?rettype=xml&study_id={phs_id}"
+        self.phs_id = phs_id
+
+    def call_telemetry_report(self):
+        """Example xml - https://www.ncbi.nlm.nih.gov/projects/gap/cgi-bin/GetSampleStatus.cgi?rettype=xml&study_id=phs000452"""
+        response = requests.get(
+            self.endpoint, 
+            headers={"Content-Type": "application/json"}
+        )
+        status_code = response.status_code
```
Comment (sahakiann) [inline src/services/dbgap_telemetry_report.py:21]: Looks like the `status_code` isn't used anywhere - Do we want to check and see if the response status was a valid one before returning the response body?

File: src/services/dbgap_telemetry_report.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+                    sra_sample_stats = sra_data["Stats"]
+
+                    if isinstance(sra_sample_stats, list):
+                        for stat in sra_sample_stats:
+                            if stat.get("@experiment_type") == data_type:
+                                return stat["@status"]
+                    elif isinstance(sra_sample_stats, dict):
+                        return sra_sample_stats["@status"]
+            
+            raise SampleNotFoundError(alias)
```
Comment (sahakiann) [inline src/services/dbgap_telemetry_report.py:44]: Maybe I'm misunderstanding, but doesn't this need to be indented one more? I think we want to raise this exception only if line 33 doesn't evaluate as true, correct?

File: src/services/dbgap_telemetry_report.py (hunk 1 above)
Comment (phendriksen100) [reply src/services/dbgap_telemetry_report.py:21]: Oh yea absolutely. Good catch!

File: src/services/dbgap_telemetry_report.py (hunk 2 above)
Comment (phendriksen100) [reply src/services/dbgap_telemetry_report.py:44]: So the for loop on line 32 goes through each sample returned from dbgap. I was thinking that this exception would be thrown if we loop through each sample and nothing is returned

Comment (sahakiann) [reply src/services/dbgap_telemetry_report.py:45]: Yes totally, ignore me! I was wrong about this

**Review by sahakiann — APPROVED** [review]

**Pull request merged**